scipy/stats/_continuous_distns.py

```diff
@@ -2263,6 +2263,90 @@ def _munp(self, n, c):
     def _entropy(self, c):
         return -_EULER / c - np.log(c) + _EULER + 1
 
+    @extend_notes_in_docstring(rv_continuous, notes="""\
+        If ``method='mm'``, parameters fixed by the user are respected, and the
+        remaining parameters are used to match distribution and sample moments
+        where possible. For example, if the user fixes the location with
+        ``floc``, the parameters will only match the distribution skewness and
+        variance to the sample skewness and variance; no attempt will be made
+        to match the means or minimize a norm of the errors.
+        \n\n""")
+    def fit(self, data, *args, **kwds):
+        if kwds.pop('superfit', False):
+            return super().fit(data, *args, **kwds)
+
+        # this extracts fixed shape, location, and scale however they
+        # are specified, and also leaves them in `kwds`
+        data, fc, floc, fscale = _check_fit_input_parameters(self, data,
+                                                             args, kwds)
+        method = kwds.get("method", "mle").lower()
+
+        # See https://en.wikipedia.org/wiki/Weibull_distribution#Moments for
+        # moment formulas.
+        def skew(c):
+            # num = 2 * gamma1**3 - 3*gamma1*gamma2 + gamma3
+            # den = (gamma2 - gamma1**2)**(3/2)
+            # Transformed to log scale due to issues on some 32-bit platforms
+            gamma1 = sc.gammaln(1+1/c)
+            gamma2 = sc.gammaln(1+2/c)
+            gamma3 = sc.gammaln(1+3/c)
+            negate = 1j*np.pi  # imaginary component of log of negative real
+            num_terms = [np.log(2) + 3*gamma1, gamma3,
+                         np.log(3) + gamma1+gamma2 + negate]
+            num = sc.logsumexp(num_terms)
+            den_terms = [gamma2, 2*gamma1 + negate]
+            den = 3/2 * sc.logsumexp(den_terms)
+            return np.real(np.exp(num-den))
```

Comment:
We were getting bad fits on some platforms. Originally, I thought it might be fixed by converting this function to the log-space, but that wasn't the problem. The real issue was that the right side of the bracket in the root search below was too high, at `c=5e5`. The `skew` as a function of `c` goes crazy for `c < 1e4` whether the calculation is done in the log-space or not. On some platforms `skew(5e5)` had the wrong sign, making the root search fail. 

![image](https://user-images.githubusercontent.com/6570539/186551834-ff3256c7-d2c6-46be-8fb6-3b39fab577ee.png)

Anyway, if it makes it easier, we can go back to the non-log version of this.

Reply:
Does the going to the log bring any benefits then? Speed, memory, accuracy? I suppose for the maintainability of the maths itself it does not matter much as formolas are available for both.

Reply:
Worse speed, worse memory. Log-scale allows smaller numbers for `c`, but it doesn't matter much because it leads to skewnesses that are infinite without log-scale:
```python3
weibull_min.stats(0.02, 0, 1, moments='s')  # 6.337061601553528e+25
weibull_min.stats(0.01, 0, 1, moments='s')  # inf
```
so I don't think it's super important. I switched back so it's easier to read.